third_party/souffle.BUILD

```diff
@@ -0,0 +1,465 @@
+        "src/include/souffle/io/gzfstream.h",
+        "src/include/souffle/io/ReadStream.h",
+        "src/include/souffle/io/ReadStreamCSV.h",
+        "src/include/souffle/io/ReadStreamJSON.h",
+        "src/include/souffle/io/ReadStreamSQLite.h",
+        "src/include/souffle/io/SerialisationStream.h",
+        "src/include/souffle/io/WriteStreamSQLite.h",
+        "src/include/souffle/io/WriteStream.h",
+        "src/include/souffle/io/WriteStreamCSV.h",
+        "src/include/souffle/io/WriteStreamJSON.h",
+        # profile headers
+        "src/include/souffle/profile/Cell.h",
+        "src/include/souffle/profile/CellInterface.h",
+        "src/include/souffle/profile/Cli.h",
+        "src/include/souffle/profile/DataComparator.h",
+        "src/include/souffle/profile/EventProcessor.h",
+        "src/include/souffle/profile/HtmlGenerator.h",
+        "src/include/souffle/profile/Iteration.h",
+        "src/include/souffle/profile/Logger.h",
+        "src/include/souffle/profile/OutputProcessor.h",
+        "src/include/souffle/profile/ProfileDatabase.h",
+        "src/include/souffle/profile/ProfileEvent.h",
+        "src/include/souffle/profile/ProgramRun.h",
+        "src/include/souffle/profile/Reader.h",
+        "src/include/souffle/profile/Relation.h",
+        "src/include/souffle/profile/Row.h",
+        "src/include/souffle/profile/Rule.h",
+        "src/include/souffle/profile/StringUtils.h",
+        "src/include/souffle/profile/Table.h",
+        "src/include/souffle/profile/Tui.h",
+        "src/include/souffle/profile/UserInputReader.h",
+        "src/include/souffle/profile/htmlCssChartist.h",
+        "src/include/souffle/profile/htmlCssStyle.h",
+        "src/include/souffle/profile/htmlJsChartistMin.h",
+        "src/include/souffle/profile/htmlJsChartistPlugin.h",
+        "src/include/souffle/profile/htmlJsMain.h",
+        "src/include/souffle/profile/htmlJsTableSort.h",
+        "src/include/souffle/profile/htmlJsUtil.h",
+        "src/include/souffle/profile/htmlMain.h",
+        # provenance headers
+        "src/include/souffle/provenance/Explain.h",
+        "src/include/souffle/provenance/ExplainProvenance.h",
+        "src/include/souffle/provenance/ExplainProvenanceImpl.h",
+        "src/include/souffle/provenance/ExplainTree.h",
+        # utility headers
+        "src/include/souffle/utility/CacheUtil.h",
+        "src/include/souffle/utility/ContainerUtil.h",
+        "src/include/souffle/utility/FileUtil.h",
+        "src/include/souffle/utility/EvaluatorUtil.h",
+        "src/include/souffle/utility/FunctionalUtil.h",
+        "src/include/souffle/utility/MiscUtil.h",
+        "src/include/souffle/utility/ParallelUtil.h",
+        "src/include/souffle/utility/StreamUtil.h",
+        "src/include/souffle/utility/StringUtil.h",
+        "src/include/souffle/utility/json11.h",
+        "src/include/souffle/utility/span.h",
+        "src/include/souffle/utility/tinyformat.h",
+        # other headers
```

Comment:
These aren't public headers and shouldn't go in `hdrs` (they belong in `srcs`).

Reply:
It *appears* as though this is pretty close to:
```
cc_library(
  srcs = glob(["src/**", exclude = ["src/include/**"]) + [":scanner", ":parser"]),
  hdrs = glob(["src/include/**"]),
  ...
)
```

Reply:
I would prefer to avoid globs (in part to possibly remove unnecessary source files in the future).

Reply:
I also need access to the internal headers.

Reply:
I've added the globs.